Assets/Scripts/DayNightCycle/TimeSwitcher.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Assets.Scripts.Services;
+using UnityEngine;
+using UnityEngine.UI;
+using Assets.Scripts.Settings;
+using UnityEngine.EventSystems;
+
+namespace Assets.Scripts.DayNightCycle
+{
+    public class TimeSwitcher : MonoBehaviour, IDragHandler
+    {
+
+        public Text Label;
+        public Slider TimeSlider;
+        public InputField TimeInput;
+        public Toggle ToggleDayNight;
+        DayAndNightControl dayNightCycle;
+
+       void Start()
+       {
+            
+            ToggleDayNight = GameObject.Find("ToggleDayNightCycle").GetComponent<Toggle>();
+            dayNightCycle = GameObject.Find("Day and Night Controller").GetComponent<DayAndNightControl>();
+             if (PhotonNetwork.isMasterClient)
+             {
+                var se = new InputField.SubmitEvent();
+                se.AddListener(SubmitTime);
+                TimeInput.onEndEdit = se;
+             }
+            
+        }
+
+
+
+
+        public void OnDrag(PointerEventData eventData)
+        {
+            if (PhotonNetwork.isMasterClient)
```

Comment:
This is a very expensive way of synchronizing time. Whenever this event is triggered, you are sending **all** settings to everyone in the lobby. Just by slowly dragging the bar I've sent +120 RPCs to the photon server. Settings should only be synchronized whenever the "Sync" button is pressed.

Each SyncSettings RPC is 5KB in data alone, so within just a send I've sent 600KB of data to the Photon Server, which it then also needs to send to every player.

Reply:
It is a tradeoff then from UX for less network use, I can put it back to just using a synch button that has to be tapped every time the slider is dragged by the user. The reason I changed it to onDrag was for UX, and I thought that once SyncSettings is expanded to only only send specific settings, so for example just time settings, that the network use would be significantly reduced. 

Alternatively a middle ground may be using OnEndDrag? That way it only synchs when the slider stops being dragged, which would be equal to what would be sent once the synch button was tapped anyway.